src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java

```diff
@@ -1,43 +1,394 @@
+      tankDriveVelocity(wheelSpeeds.leftMetersPerSecond, wheelSpeeds.rightMetersPerSecond);
+    } else {
+      leftMaster.set(FEED_FORWARD.calculate(wheelSpeeds.leftMetersPerSecond) / 12);
+      rightMaster.set(FEED_FORWARD.calculate(wheelSpeeds.rightMetersPerSecond) / 12);
+    }
+  }
+
+  /**
+   * Drives the robot by individually addressing the left and right side of the
+   * drive train
+   * 
+   * @param leftSpeed  speed of the left motors [-1.0..1.0]
+   * @param rightSpeed speed of the right motors [-1.0..1.0]
+   * @param useSquares if set, decreases input sensitivity at low speeds
+   */
+  public void tankDrive(double leftSpeed, double rightSpeed, boolean useSquares) {
+    var xLeftSpeed = safeClamp(leftSpeed) * MAX_SPEED_ARCADE;
+    var xRightSpeed = safeClamp(rightSpeed) * MAX_SPEED_ARCADE;
+    if (useSquares) {
+      xLeftSpeed *= Math.abs(xLeftSpeed);
+      xRightSpeed *= Math.abs(xRightSpeed);
+    }
+    if(useEncoders) {
+      tankDriveVelocity(xLeftSpeed, xRightSpeed);
+    } else {
+      tankDriveRaw(FEED_FORWARD.calculate(xLeftSpeed) / 12, FEED_FORWARD.calculate(xRightSpeed) / 12);
+    }
+  }
+
+  /**
+   * WARNING this method doesn't used encoders, squaring, or feed forward logic. Use at your own risk!
+   * @param leftSpeed
+   * @param rightSpeed
+   */
+  public void tankDriveRaw(double leftSpeed, double rightSpeed) {
+    leftMaster.set(leftSpeed);
+    rightMaster.set(rightSpeed);
+  }
+
+  /**
+   * Controls the left and right side of the drive using Talon SRX closed-loop
+   * velocity.
+   * 
+   * @param leftVelocity  left velocity in meters per second
+   * @param rightVelocity right velocity in meters per second
+   */
+  public void tankDriveVelocity(double leftVelocity, double rightVelocity) {
+    var leftFeedForwardVolts = FEED_FORWARD.calculate(leftVelocity);
+    var rightFeedForwardVolts = FEED_FORWARD.calculate(rightVelocity);
+
+    leftMaster.set(
+        ControlMode.Velocity, 
+        metersPerSecToEdgesPerDecisec(leftVelocity), 
+        DemandType.ArbitraryFeedForward,
+        leftFeedForwardVolts / 12);
+    rightMaster.set(
+        ControlMode.Velocity,
+        metersPerSecToEdgesPerDecisec(rightVelocity),
+        DemandType.ArbitraryFeedForward,
+        rightFeedForwardVolts / 12);
+  }
+
+  /**
+   * Sets the drivetrain to zero velocity and rotation.
+   */
+  public void stop() {
+    leftMaster.set(0);
+    rightMaster.set(0);
+    speedRateLimiter.reset(0);
+    rotationRateLimiter.reset(0);
+  }
+
+  /**
+   * Returns value clamped between [-1, 1]. Not-a-number (NaN) returns 0;
+   * @param input value to clamp
+   * @return clamped value
+   */
+  private double safeClamp(double input) {
+    if (Double.isNaN(input)) {
+      return 0;
+    }
+    return MathUtil.clamp(input, -1, 1);
+  }
+
+  /**
+   * Sets the neutral mode for the drive train
+   * 
+   * @param neutralMode the desired neutral mode
+   */
+  public void setNeutralMode(NeutralMode neutralMode) {
```

Comment:
It would be nice to use this method to put the motors in coast mode when the robot is disabled. See how we did this with a `Trigger` on the [tank bot](https://github.com/STMARobotics/paradebot-7028/blob/master/src/main/java/frc/robot/subsystems/DriveTrainSubsystem.java#L78)